pkg/metrics/controller_metrics.go

```diff
@@ -21,6 +21,7 @@ import (
 )
 
 var (
+	distributionSeconds       = []float64{0, 0.01, 0.025, 0.05, 0.075, 0.1, 0.25, 0.5, 0.75, 1, 2, 3}
```

Comment:
Not sure about those buckets. What are your expectations from a range of min to max duration for each state ?

I have a feeling this could go way beyond 3 seconds, since this is a histogram you're going to have an average and quantile very wrong.

Might even needs to be configurable by the users with a sane default. Let's start with a just a sane default for now. This is a game of precisions vs memory usage, I would have more than 20buckets for instance.

Reply:
You are right, buckets boundaries are unusual for state duration.